spec/hotel_spec.rb

```diff
@@ -0,0 +1,80 @@
+require_relative 'spec_helper'
+
+describe "hotel" do
+  NUM_BOOKED_ROOMS = Hotel::NUM_ROOMS - 5
+
+  before do
+    @seventh = Time.parse("2018-07-07")
+    @eleventh = Time.parse("2018-07-11")
+  end
+
+  let(:empty_hotel) do
+    empty_hotel = Hotel.new
+  end
+
+  let(:booked_hotel) do
+    booked_hotel = Hotel.new
+    (NUM_BOOKED_ROOMS).times do
+      booked_hotel.make_reservation(@seventh, @eleventh)
+    end
+    booked_hotel
+  end
+
+  let(:fully_booked_hotel) do
+    fully_booked_hotel = Hotel.new
+    Hotel::NUM_ROOMS.times do
+      fully_booked_hotel.make_reservation(@seventh, @eleventh)
+    end
+    fully_booked_hotel
+  end
+
+  it "get rooms" do
+    rooms = empty_hotel.get_rooms
+    expect(rooms.length).must_equal Hotel::NUM_ROOMS
+  end
+
+  describe "makes reservation" do
+    it "make reservation for hotel with no reservations" do
+      reservation = empty_hotel.make_reservation(@seventh, @eleventh)
+    end
+
+    it "make reservation for hotel with several reservations" do
+      reservation = booked_hotel.make_reservation(@seventh, @eleventh)
+    end
+
+    it "make reservation for fully booked hotel" do
+      expect{fully_booked_hotel.make_reservation(@seventh, @eleventh)}.must_raise RuntimeError
+    end
```

Comment:
There are a number of questions around when specifically a room is available that these tests don't cover. In particular, if a hotel is fully booked for one range of dates and you get another request...

No room is available if the dates:

- Same dates
- Overlaps in the front
- Overlaps in the back
- Completely contained
- Completely containing

The room is available if the dates:

- Completely before
- Completely after
- Ends on the checkin date
- Starts on the checkout date